Client/Frontend/Shields/ShieldsActivityItemSourceProvider.swift

```diff
@@ -0,0 +1,127 @@
+    case instagram = "com.burbn.instagram.shareextension"
+}
+
+// MARK: - ShieldsActivityItemSourceProvider
+
+final class ShieldsActivityItemSourceProvider {
+    
+    static let shared = ShieldsActivityItemSourceProvider()
+    
+    func setupGlobalShieldsActivityController(theme: Theme) -> UIActivityViewController {
+        let backgroundImage = #imageLiteral(resourceName: "share-activity-background")
+        
+        let statsView = UIView(frame: CGRect(size: backgroundImage.size)).then {
+            let backgroundImageView = UIImageView(image: backgroundImage)
+            let statsInfoView = BraveShieldStatsView().then {
+                $0.applyTheme(theme)
+            }
+            
+            $0.addSubview(backgroundImageView)
+            $0.addSubview(statsInfoView)
+            
+            backgroundImageView.snp.makeConstraints {
+                $0.edges.equalToSuperview()
+            }
+            statsInfoView.snp.makeConstraints {
+                $0.centerX.equalToSuperview()
+                $0.centerY.equalToSuperview().offset(15)
+                $0.height.equalToSuperview().multipliedBy(0.40)
+                $0.width.equalToSuperview().multipliedBy(0.70)
+            }
+        }
+    
+        let contentView = UIView(frame: CGRect(width: statsView.frame.width, height: statsView.frame.height + 85)).then {
+            $0.backgroundColor = UIColor(rgb: 0x4C54D2)
+            $0.layer.borderWidth = 1
+        }
+        
+        contentView.addSubview(statsView)
+        statsView.frame = CGRect(origin: .zero,
+                                 size: CGSize(width: statsView.frame.width, height: statsView.frame.height))
+        
+        let snapshotImage = statsView.snapshot
+        let snapshotImageWithText = contentView.snapshot.textToImage(drawText: Strings.ShieldEducation.shareDescriptionTitle,
+                                                                     atPoint: CGPoint(x: 0, y: statsView.frame.height + 20)) ?? snapshotImage
+
+        let activityViewController = UIActivityViewController(activityItems: [ImageActivityItemSource(image: snapshotImage,
+                                                                                                      imageWithText: snapshotImageWithText),
+                                                                              OptionalTextActivityItemSource(text: Strings.ShieldEducation.shareDescriptionTitle)],
+                                                              applicationActivities: nil)
+        
+        activityViewController.excludedActivityTypes = [.openInIBooks, .saveToCameraRoll, .assignToContact]
+        
+        return activityViewController
+    }
+    
+}
+
+// MARK: - OptionalTextActivityItemSource
+
+class OptionalTextActivityItemSource: NSObject, UIActivityItemSource {
+    
+    let text: String
+    
+    weak var viewController: UIViewController?
+    
+    init(text: String) {
+        self.text = text
```

Comment:
Missing call to `super.init()` when inheriting `NSObject`

Reply:
Fix at https://github.com/brave/brave-ios/pull/3387/commits/da11867eb41296b14bf62ec79fb7344b5577df37